server/src/__tests__/analyzer.test.ts

```diff
@@ -0,0 +1,106 @@
+import { generateParser } from '../tree-sitter/parser'
+import Analyzer from '../tree-sitter/analyzer'
+import { FIXTURE_DOCUMENT } from './fixtures/fixtures'
+
+// Needed as the param
+const DUMMY_URI = 'dummy_uri'
+
+async function getAnalyzer (): Promise<Analyzer> {
+  const parser = await generateParser()
+  const analyzer = new Analyzer()
+  analyzer.initialize(parser)
+  return analyzer
+}
+
+const initialize = jest.spyOn(Analyzer.prototype, 'initialize')
+const wordAtPoint = jest.spyOn(Analyzer.prototype, 'wordAtPoint')
+
+describe('analyze', () => {
+  it('instantiates an analyzer', async () => {
+    // Alternative: Spy on something (logger) within the analyzer instead of spying on every function in the Analyzer
+    await getAnalyzer()
+
+    expect(initialize).toHaveBeenCalled()
+  })
+
+  it('analyzes simple correct bb file', async () => {
+    const analyzer = await getAnalyzer()
+    const diagnostics = await analyzer.analyze({
+      uri: DUMMY_URI,
+      document: FIXTURE_DOCUMENT.CORRECT
+    })
+    expect(diagnostics).toEqual([])
+  })
+
+  it('analyzes the document and returns global declarations', async () => {
+    const analyzer = await getAnalyzer()
+    await analyzer.analyze({
+      uri: DUMMY_URI,
+      document: FIXTURE_DOCUMENT.DECLARATION
+    })
+
+    const globalDeclarations = analyzer.getGlobalDeclarationSymbols(DUMMY_URI)
+
+    expect(globalDeclarations).toMatchInlineSnapshot(`
```

Comment:
This works but as we add more tests, wouldn't it be more structured to define all test cases by having a file and a json truth?
```
__tests__/analyzer/fixtures/dummy.bb
__tests__/analyzer/fixtures/dummy.bb.json
```

Reply:
what would you have in this JSON? And there isn't a programmatic way to provide snapshots of the tests right now.